hpa/README.md

```diff
@@ -0,0 +1,76 @@
+
+The `experiment.yaml` file is the actual experiment object manifest; this includes the definition of the experiment itself (in terms of assignable parameters and observable metrics) and the instructions for carrying out the experiment (in terms of load generation, configured in `trialjob_configmap.yaml`, and metric queries). Here is where you may change parameter ranges and the experiment name (to avoid experiment name conflicts in the cluster).
+
+## Experiment lifecycle
+
+For each trial, we create a locust load test for the configuration tested using the following trial pod:
+
+```
+          template: # pod
+            spec:
+              containers:
+              - name: locust-trial-job
+                image: broadinstitute/python-requests
+                command: ["python", "/home/load_test.py"]
+                env:
+                - name: LOCUST_HOST
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: locust_host
+                - name: N_CLIENTS
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: n_clients
+                - name: HATCH_RATE
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: hatch_rate
+                - name: RUN_TIME
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: run_time
+                - name: MAX_LATENCY # max avg latency in ms
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: max_latency
+                - name: MAX_ERROR_RATE # max error rate in %
+                  valueFrom:
+                    configMapKeyRef:
+                      name: trialjob-env
+                      key: max_error_rate
+```
+
+The configuration for the load test is in `trialjob_configmap.yaml`. The load test is 180 seconds long, uses 100 clients that are added at a rate of 1 client/second. You can increase this rate to make sure that your HPA scales fast enough to increase in load.
+[However, with a kubernetes version lower than `v1.18` you cannot change the scaling policies through the `v2beta2` API](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/). In this experiment, we are going to optimize for the `target CPU utilization` that the HPA uses to scale the `voting-service`  deployment.
+
+`MAX_LATENCY` is the maximum latency in ms we want a trial to report. If locust measures an average latency over the length of the load test (aka trial) higher than this value, we will report this trial as failed. `MAX_ERROR_RATE` is the maximum error rate in % measured by locust for which we don't report the trial as failed. You can see the logic for this in `load_test.py`.
```

Comment:
The error rate portion is a little confusing to me, but it could be because I'm not the right audience. Looking at load_test.py it looks like error rate =numfailures/numrequests*100 and the max error rate is your tolerance for the trial. If I'm interpreting that correctly maybe:
```suggestion
`MAX_LATENCY` is the maximum latency (in milliseconds) we want a trial to report. If locust measures an average latency over the length of the load test (aka trial) higher than this value, we will report this trial as failed. Similarly, `MAX_ERROR_RATE` is the maximum error rate (reported as %) we want a trial to report. An error rate higher than this value will fail the trial. You can see the logic for this in `load_test.py`.
 ```